File: examples/canonical-iac-pulumi/dotnet/azure/components/storage-component/StorageComponent.cs
using Pulumi;
using Pulumi.AzureNative.Storage;

public sealed class StorageComponent : ComponentResource
{
    public Output<string> StorageAccountName { get; private set; }

    public StorageComponent(string name, string environment, string namePrefix, ComponentResourceOptions? opts = null)
        : base("accb:azure:StorageComponent", name, opts)
    {
        var stackName = Deployment.Instance.StackName;
        var account = new StorageAccount($"{namePrefix}-{environment}-{stackName}-st", new StorageAccountArgs
        {
            Kind = Kind.StorageV2,
            Sku = new Pulumi.AzureNative.Storage.Inputs.SkuArgs { Name = SkuName.Standard_LRS }
        }, new CustomResourceOptions { Parent = this });
        StorageAccountName = account.Name;
        RegisterOutputs(new Dictionary<string, object?> { ["storageAccountName"] = StorageAccountName });
    }
}

File: examples/canonical-iac-pulumi/python/azure/components/storage_component.py
import pulumi


class StorageComponent(pulumi.ComponentResource):
    def __init__(self, name: str, environment: str, name_prefix: str, opts=None):
        super().__init__("accb:azure:StorageComponent", name, None, opts)
        stack = pulumi.get_stack()
        self.storage_name = pulumi.Output.from_input(f"{name_prefix}-{environment}-{stack}-artifacts")
        self.register_outputs({"storage_name": self.storage_name})
